### ISSUE TITLE
RFC: Deprecate Bottleneck

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#68742]
While discussing https://github.com/pytorch/pytorch/issues/68433 there appear to be no owners for Bottleneck and it seems to be unmaintained. If this is the case and we are encouraging users to use newer profiling methods we should consider deprecating it ahead of the PyTorch 1.11 release.

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @bdhirsh @jbschlosser @VitalyFedyunin @ngimel @ilia-cher @robieta @chaekit @gdankel @bitfort @orionr @nbcsm @guotuofeng @guyang3532 @gaoteng-git

**Comment by janeyx99** [comment]
@mruberry From the response to this issue, it looks like we should be safe to deprecate it now, yes?

**Comment by mruberry** [comment]
@janeyx99 We should get its creators to comment before proceeding

**Comment by janeyx99** [comment]
@mruberry Who are the creators?

**Comment by mruberry** [comment]
@zou3519?

**Comment by zou3519** [comment]
I agree it should be deprecated in favor of the new profiling techniques. I don't know what the recommended API to use now is so I won't be very helpful in writing a deprecation message. 

A cursory search through GitHub shows that there are folks using it: https://github.com/search?l=Python&p=2&q=torch.utils.bottleneck&type=Code so we should do a deprecation cycle instead of removing it entirely in a single release.

**Comment by sraikund16** [comment]
Removing oncall: profiler label as there does not seem to be any instances of bottleneck in the profiler code
